File: solutions/c/armstrong-numbers/1/armstrong_numbers.c
#include "armstrong_numbers.h"
#include <math.h>

bool is_armstrong_number(int candidate) {
    int n = candidate;
    int sum = 0;
    int len = 0;

    while (n > 0) {
        n /= 10;
        len++;
    }
    n = candidate;

    while (n > 0) {
        sum += pow(n % 10, len);
        n /= 10;
    }
    return sum == candidate;
}

File: solutions/php/armstrong-numbers/1/ArmstrongNumbers.php
<?php
declare(strict_types=1);

function isArmstrongNumber(int $number): bool
{
    $sum = 0;
    $len = strlen((string) $number);

    for ($i = 0; $i < $len; $i++) {
        $sum += pow((int) substr((string) $number, $i, 1), $len);
    }

    return $sum === $number;
}
